lib/mp-readline/readline.c

```diff
@@ -98,6 +99,38 @@ typedef struct _readline_t {
 
 STATIC readline_t rl;
 
+size_t get_previous_word_cursor_pos() {
+  size_t cursor_pos = rl.cursor_pos;
+  bool in_leading_nonalphanum = true;
+  while (cursor_pos > 0) {
+    if (!isalnum((unsigned char) rl.line->buf[cursor_pos - 1])) {
```

Comment:
Please see `py/unicode.c` for functions that can be used to test character sets.  We can't rely on ctype.h.

Reply:
Now using `unichar_is*` funcs: https://github.com/micropython/micropython/pull/5024/commits/00b2a0e2609ed84df5be05af728c551c457e622f#diff-ec323dc5c7454143942798519678d424R113